tests/test_wagon.py

```diff
@@ -0,0 +1,686 @@
+        finally:
+            shutil.rmtree(tmpdir)
+
+    def test_tar_missing_source(self):
+        e = self.assertRaises(OSError, wagon._tar, 'missing', 'file')
+        if wagon.IS_WIN:
+            self.assertIn(
+                "The system cannot find the file specified: 'missing'",
+                str(e))
+        else:
+            self.assertIn("No such file or directory: 'missing'", str(e))
+        os.remove('file')
+
+    def test_make_virtualenv_illegal_dir(self):
+        if wagon.IS_WIN:
+            self.skipTest('Irrelevant on Windows')
+        illegal_path = '/opt/test_virtualenv'
+        ex = self.assertRaises(
+            wagon.WagonError,
+            wagon._make_virtualenv,
+            illegal_path)
+        self.assertIn('Could not create virtualenv', str(ex))
+
+    def test_wheel_nonexisting_package(self):
+        try:
+            e = self.assertRaises(
+                wagon.WagonError, wagon.wheel, 'cloudify-script-plug==1.3')
+            self.assertIn('Could not download wheels for:', str(e))
+        finally:
+            shutil.rmtree('package')
+
+    def test_machine_platform(self):
+        self.assertIn(
+            'win32' if wagon.IS_WIN else 'linux_x86_64',
+            wagon.get_platform())
+
+    def test_get_version_from_pypi_bad_source(self):
+        e = self.assertRaises(
+            wagon.WagonError,
+            wagon._get_package_info_from_pypi,
+            'NONEXISTING_PACKAGE')
+        self.assertIn('Failed to retrieve info for package', str(e))
+
+    def test_check_package_not_installed(self):
+        wagon._make_virtualenv('test_env')
+        try:
+            result = wagon._check_installed(TEST_PACKAGE_NAME, 'test_env')
+            self.assertFalse(result)
+        finally:
+            shutil.rmtree('test_env')
+
+    def test_install_package_failed(self):
+        e = self.assertRaises(
+            wagon.WagonError, wagon.install_package, 'x', 'y')
+        self.assertIn('Could not install package:', str(e))
+
+    def test_archive_unsupported_format(self):
+        e = self.assertRaises(
+            wagon.WagonError,
+            wagon._create_wagon_archive,
+            'unsupported_format',
+            'source_dir',
+            'output_archive')
+        self.assertIn('Unsupported archive format to create', str(e))
+
+    def test_single_python_version(self):
+        versions = wagon._set_python_versions()
+        verinfo = sys.version_info
+        version = ['py{0}{1}'.format(verinfo[0], verinfo[1])]
+        self.assertEqual(versions, version)
+
+    def test_provided_python_version(self):
+        provided_version_numbers = ['27', '26']
+        versions = wagon._set_python_versions(provided_version_numbers)
+        expected_versions = ['py27', 'py26']
+        self.assertEqual(versions, expected_versions)
+
+    @mock.patch('sys.executable', new='/a/b/c/python')
+    def test_pip_path_on_linux(self):
+        if wagon.IS_WIN:
```

Comment:
to make the test code cleaner take advantage of testtools.skipIf decorator. like so:
@testtools.skipIf(wagon.IS_WIN, 'Irrelevant on Windows.')